lib/example_17x_app_web/views/thing_view.ex

```diff
@@ -1,10 +1,19 @@
-defmodule Example17xAppWeb.ThingView do
-  use Example17xAppWeb, :view
+# Using an erlang atom as module name, otherwise dashes would not be allowed in the module name.
+# And if I don't do this then I get the error:
+#
+#   (ArgumentError) no "show" json-api template defined for
+#   :"Elixir.Example17xAppWeb.ThingJSON-API"  (the module does not exist)
+#
+defmodule :"Elixir.Example17xAppWeb.ThingJSON-API" do
```

Comment:
So (I suspect that) because of the mime type / format "json-api" changes I had to make in this PR, along with the way that phoenix 1.7.7 renders views, I would get the following error if I used the pre-phoenix 1.7.7 module name style `Example17xAppWeb.ThingView`...

```
(ArgumentError) no "show" json-api template defined for :"Elixir.Example17xAppWeb.ThingJSON-API"  (the module does not exist)
```

If I did not define the `type()` function below then, when I run the test suite, I would get...

![image](https://github.com/theirishpenguin/example_17x_app/assets/39653/799e41e8-0c59-4177-a650-fc4c846f12c9)

I am speculating there is code in JA serializer that...

* Takes the view name ThingView
* Removes "View" from the end
* And uses this as the default "type"
* But this now longer works since the module name ends in "JSON-API"
* (Reminder: This is speculation. I could be wrong. But something funny is happening anyway)

Reply:
This makes sense because it infers the "type" from the module name.